plugin/src/main/java/org/opensearch/ml/action/tasks/CancelBatchJobTransportAction.java

```diff
@@ -0,0 +1,226 @@
+                    ActionListener<Connector> listener = ActionListener
+                        .wrap(connector -> { executeConnector(connector, mlInput, actionListener); }, e -> {
+                            log.error("Failed to get connector " + model.getConnectorId(), e);
+                            actionListener.onFailure(e);
+                        });
+                    try (ThreadContext.StoredContext threadContext = client.threadPool().getThreadContext().stashContext()) {
+                        connectorAccessControlHelper
+                            .getConnector(client, model.getConnectorId(), ActionListener.runBefore(listener, threadContext::restore));
+                    }
+                } else {
+                    actionListener.onFailure(new ResourceNotFoundException("Can't find connector " + model.getConnectorId()));
+                }
+            }, e -> {
+                log.error("Failed to retrieve the ML model with the given ID", e);
+                actionListener.onFailure(e);
+            }));
+        } catch (Exception e) {
+            // fetch the connector
+            log.error("Unable to fetch status for ml task ", e);
+        }
+    }
+
+    private void executeConnector(Connector connector, MLInput mlInput, ActionListener<MLCancelBatchJobResponse> actionListener) {
+        connector.decrypt(BATCH_PREDICT_STATUS.name(), (credential) -> encryptor.decrypt(credential));
+        RemoteConnectorExecutor connectorExecutor = MLEngineClassLoader.initInstance(connector.getProtocol(), connector, Connector.class);
+        connectorExecutor.setScriptService(scriptService);
+        connectorExecutor.setClusterService(clusterService);
+        connectorExecutor.setClient(client);
+        connectorExecutor.setXContentRegistry(xContentRegistry);
+        connectorExecutor.executeAction(BATCH_PREDICT_STATUS.name(), mlInput, ActionListener.wrap(taskResponse -> {
+            processTaskResponse(taskResponse, actionListener);
+        }, e -> { actionListener.onFailure(e); }));
+    }
+
+    private void processTaskResponse(MLTaskResponse taskResponse, ActionListener<MLCancelBatchJobResponse> actionListener) {
+        try {
+            ModelTensorOutput tensorOutput = (ModelTensorOutput) taskResponse.getOutput();
+            if (tensorOutput != null && tensorOutput.getMlModelOutputs() != null && !tensorOutput.getMlModelOutputs().isEmpty()) {
+                ModelTensors modelOutput = tensorOutput.getMlModelOutputs().get(0);
+                if (modelOutput.getStatusCode() != null && modelOutput.getStatusCode().equals(HttpStatus.SC_OK)) {
+                    actionListener.onResponse(new MLCancelBatchJobResponse(RestStatus.OK));
+                } else {
+                    log.debug("The status code from remote service is: " + modelOutput.getStatusCode());
+                    actionListener.onFailure(new ResourceNotFoundException("Couldn't cancel the transform job. Please try again"));
```

Comment:
Why use ResourceNotFoundException here? It seems like that it can be any Exception from remote service. So we can just actionListener.onFailure(new OpenSearchException(modelOutput.getStatusCode()));

Reply:
Yeah makes sense